frontend/src/components/utils/file_upload.tsx

```diff
@@ -2,32 +2,47 @@ import { Group, Text } from '@mantine/core';
 import { Dropzone, MIME_TYPES } from '@mantine/dropzone';
 import { IconCloudUpload, IconDownload, IconX } from '@tabler/icons-react';
 import { useTranslation } from 'next-i18next';
-import { useRef } from 'react';
+import { useMemo, useRef } from 'react';
 import { SWRResponse } from 'swr';
 
+import { AxiosError } from 'axios';
 import { Tournament } from '../../interfaces/tournament';
-import { handleRequestError, uploadLogo } from '../../services/adapter';
+import { handleRequestError, uploadTeamLogo, uploadTournamentLogo } from '../../services/adapter';
+import { TeamInterface } from '../../interfaces/team';
 
 export function DropzoneButton({
-  tournament,
-  swrTournamentResponse,
+  tournamentId,
+  swrResponse,
+  variant,
+  teamId,
 }: {
-  tournament: Tournament;
-  swrTournamentResponse: SWRResponse;
+  tournamentId: Tournament['id'];
+  swrResponse: SWRResponse;
+  variant: 'tournament' | 'team';
+  teamId?: TeamInterface['id'];
 }) {
   // const { classes, theme } = useStyles();
   const openRef = useRef<() => void>(null);
   const { t } = useTranslation();
 
+  const useUploadLogo = useMemo(() => {
+    if (variant === 'tournament') {
+      return uploadTournamentLogo.bind(null, tournamentId);
+    }
+
+    if (teamId === undefined) throw new TypeError('Team is undefined');
+    return uploadTeamLogo.bind(null, tournamentId, teamId);
+  }, [tournamentId, teamId, variant]);
+
   return (
     <div>
       <Dropzone
         mt="lg"
         openRef={openRef}
         onDrop={async (files) => {
-          const response = await uploadLogo(tournament.id, files[0]);
-          await swrTournamentResponse.mutate();
-          handleRequestError(response);
+          const response = await useUploadLogo(files[0]);
+          await swrResponse.mutate();
+          handleRequestError(response as unknown as AxiosError); // TODO: Check with Erik if this is correct
```

Comment:
Nope, TypeScript complains about the insufficient overlap. I just added type defs to the axios library as it was annoying me that there weren't any. Also why is Axios required rather than imported?

Reply:
> Also why is Axios required rather than imported?

Ah no particular reason, must be an oversight.